[BUGFIX release] Fix container destroy timing

wagenet opened the pull request.

wagenet committed:
Fix container destroy timing

rwjblue reviewed (changes requested):
Need to tweak some other parts of the container destroy lifecycle also here, I'll get those changes pushed up into this PR shortly...

stefanpenner commented on packages/ember-runtime/lib/mixins/container_proxy.js:
```diff
@@ -101,7 +101,7 @@ let containerProxyMixin = {
 
     if (this.__container__) {
-      run(this.__container__, 'destroy');
+      schedule('destroy', this.__container__, 'destroy');
```
I think more `join` like behavior?

wagenet committed:
Fix code style

Commit:
[BUGFIX release] Refactor owner/container destroy.

Ensure `.destroy()` is called on all items in the container, _before_
marking `container.isDestroyed`. Only flush the caches after destroy is
finalized.

rwjblue commented:
Just pushed an update that should address the concerns that @krisselden / @stefanpenner brought up.

Should be good to land once green...

wagenet reviewed (commented):
Looks ok. Only point of confusion I had is that `finalizeDestroy` isn't automatically called. Is that as expected? Obviously, this is internal API, but I could see someone potentially calling only `destroy` and never calling `finalizeDestroy`.

rwjblue commented:
> Only point of confusion I had is that finalizeDestroy isn't automatically called. Is that as expected? Obviously, this is internal API, but I could see someone potentially calling only destroy and never calling finalizeDestroy.

@wagenet - The container itself is not actually exposed, any user-land code calling `.destroy()` would be on the "owner" which is exactly the thing that mixes in `ContainerProxyMixin` and gets the `destroy` implementation in this PR.
